extractor/src/main/java/org/schabi/newpipe/extractor/search/SearchExtractor.java

```diff
@@ -31,7 +28,7 @@ public String getSearchString() {
     public abstract String getSearchSuggestion() throws ParsingException;
 
     protected InfoItemsSearchCollector getInfoItemSearchCollector() {
-        return collector;
+        return new InfoItemsSearchCollector(getService().getServiceId());
```

Comment:
Do we really need a `getInfoItemSearchCollector()` function? We don't have one in the other kind of extractors either. Otherwise this PR looks good, however you'll need to do YouTube Music tests as well when that PR gets merged.